### PR TITLE
Improve the docs for upgrading from SDKv2 to a muxed provider

### LINKED ISSUE #1690: Improve documentation for upgrading from a SDKv2 to a muxed provider
From walking a community member through upgrading, we need to:
1. Include import statements in examples: we use `pfbridge` and `tfbridge` without explaining what those imports mean.
2. Tell a more complete store about which bits are library code and which bits are user provided. `provShim.PFProvider()` should be explicitly called out as external in the example.
3. That metadata info is needed should be explicitly called out in a more aggressive way.
4. ☝️ should refer to a `docs/metadatainfo.md` card that explains what it is and how to add it.

### PR DESCRIPTION
**Pull request by iwahbe**
Fixes #1690

This PR adds clarity and cross-linking to our "upgrade to muxed" docs.

### CODE REVIEW

File: docs/upgrade-sdk-to-mux.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+Follow these steps if you have a Pulumi provider that was bridged from a Terraform
+provider built against [Terraform Plugin
+SDK](https://github.com/hashicorp/terraform-plugin-sdk) and you want to upgrade it to a
+version that has migrated some but not all resources/datasources to the [Plugin
+Framework](https://github.com/hashicorp/terraform-plugin-sdk?tab=readme-ov-file).
+
+1. Ensure you have access to the
+   `github.com/hashicorp/terraform-plugin-framework/provider.Provider` from the upstream
+   provider.  If the provider is shimmed (or needs to be), you can follow step (1) from
+   ["How to Upgrade a Bridged Provider to Plugin Framework"](../pf/README.md#How to
```
Comment (VenelinMartinov) [inline docs/upgrade-sdk-to-mux.md:12]: is the link here broken? the review does not render it proerly

File: docs/upgrade-sdk-to-mux.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+   ``` go
+   import "github.com/pulumi/pulumi-terraform-bridge/pf/tfgen"
+
+   ...
+
+   tfgen.MainWithMuxer("cloudflare", cloudflare.Provider())
+   ```
+
+1. Find the provider binary `main` that calls
```
Comment (VenelinMartinov) [inline docs/upgrade-sdk-to-mux.md:39]: the numbers are all 1.

File: docs/upgrade-sdk-to-mux.md (hunk 1 above)
Comment (iwahbe) [reply docs/upgrade-sdk-to-mux.md]: Fixed.

File: docs/upgrade-sdk-to-mux.md (hunk 2 above)
Comment (VenelinMartinov) [reply docs/upgrade-sdk-to-mux.md:39]: TIL

**Review by VenelinMartinov — APPROVED** [review]

**Comment by VenelinMartinov** [comment]
Ah, you've hit https://github.com/pulumi/pulumi/issues/14240

**Pull request merged**